02_bubble/joshcarp/bubble_test.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var stdout = os.Stdout
+
+var tests = []struct {
+	input    []int
+	expected []int
+}{
+	{[]int{3, 2, 1, 5}, []int{1, 2, 3, 5}},
+	{[]int{-71, -99, 89, 4, -82, -73, -65, 94, -92, 99}, []int{-99, -92, -82, -73, -71, -65, 4, 89, 94, 99}},
+}
+
+func TestBubble(t *testing.T) {
+	for _, test := range tests {
+		calculated := bubble(test.input)
+		require.Equal(t, test.expected, calculated)
```

Comment:
Why use `require` instead of `assert`? 
I think `require` will ignore the rest test cases once a test case fails.